src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -150,6 +213,237 @@ def get_S3FileSystem_from_session(
+        non_empty_contents = [
+            f.filename
+            for f in z.filelist
+            if "/" not in f.filename
+            and "Manifest" not in f.filename
+            and f.file_size > 0
+        ]
+        return non_empty_contents
+
+
+def get_integration_test_exports_json(
+    s3: boto3.client, cfn_bucket: str, staging_namespace: str
+) -> List[str]:
+    """Reads in the integration test exports json from the cloudformation
+        bucket that contains the ~2 weeks of production data exports
+        to use in the staging pipeline
+
+    Args:
+        s3 (boto3.client): s3 client connection
+        cfn_bucket (str): cloudformation bucket name
+        staging_namespace (str): name of namespace containing the "new" data
+
+    Returns:
+        List[str]: list of the json exports
+    """
+    # read in the json filelist
+    s3_response_object = s3.get_object(
+        Bucket=cfn_bucket, Key=f"{staging_namespace}/integration_test_exports.json"
+    )
+    json_content = s3_response_object["Body"].read().decode("utf-8")
+    filelist = json.loads(json_content)
+    return filelist
+
+
+def get_exports_filter_values(
+    s3: boto3.client,
+    data_type: str,
+    input_bucket: str,
+    cfn_bucket: str,
+    staging_namespace: str,
+) -> ds.Expression:
+    """Parses through the json exports and gets the filter values
+    for the cohort and export_end_date to filter on for our
+    main parquet dataset. These filter values (dict) are then
+    converted into filter conditions in the form of a
+    pyarrow.dataset.Expression object
+
+    Args:
+        s3 (boto3.client): s3 client connection
+        data_type (str): data type of the dataset
+        input_bucket (str): input data bucket name
+        cfn_bucket (str): cloudformation bucket name
+        staging_namespace (str): name of namespace containing the "new" data
+
+    Returns:
+        ds.Expression: a pyarrow dataset expression object that contains
+        the filter conditions that can be applied to pyarrow datasets
+    """
+    filelist = get_integration_test_exports_json(s3, cfn_bucket, staging_namespace)
+    # create the dictionary of export end dates and cohort
+    export_end_date_vals = {}
+    for s3_uri in filelist:
+        json_files = get_json_files_in_zip_from_s3(s3, input_bucket, s3_uri)
+        cur_cohort = get_cohort_from_s3_uri(s3_uri)
+        for json_file in json_files:
+            cur_data_type = get_data_type_from_filename(json_file)
+            cur_export_end_date = get_export_end_date(json_file)
+            if cur_data_type == data_type:
+                if cur_cohort in export_end_date_vals.keys():
+                    export_end_date_vals[cur_cohort].append(cur_export_end_date)
+                else:
+                    export_end_date_vals[cur_cohort] = [cur_export_end_date]
+            else:
+                continue
+
+    # create the filter using pyarrow and predicate pushdown
```

Comment:
Actually yes i forgot this, I could just replace `get_parquet_dataset` with `get_filtered_main_dataset` because if `filter` is None in `get_filtered_main_dataset`, it will just read in the dataset still as a whole with no filter.

Yes originally when I first designed this function, I wrestled between splitting the current `get_exports_filter_values` function into the two you described mainly because I felt like the tests would be sort of redundant between the two because our end goal of the function is that it returns the right ds.Expression (filter and values) (and we already have so many tests). But i can see a use case where we may want to utilize the dict or list[namedtuple] of filter values over a `ds.Expression` object so I'll just split the two functions. Easier to troubleshoot as well in the future.